### PR TITLE
feat: generic calculations

### PR DESCRIPTION
**Pull request by Xazin**
This PR adds Count, CountEmpty, and CountNonEmpty as calculations, and these are supported on all Field Types where empty values are allowed. (Count is supported on all field types).

I also removed that little "Count: #" at the bottom of the grid, as the same can now be achieved by adding a calculation.

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/rust-lib/flowy-database2/src/services/calculations/controller.rs [hunk 1]
```diff
@@ -215,19 +217,34 @@ impl CalculationsController {
...
+      if let Some(view_id) = view_id {
+        let calculations = self.delegate.get_all_calculations(&view_id).await;
+        for calculation in calculations.iter() {
+          let update = self.get_updated_calculation(calculation.clone()).await;
+          if let Some(update) = update {
+            updates.push(CalculationPB::from(&update));
+            self.delegate.update_calculation(&view_id, update);
+          }
+        }
+      }
```
Comment (richardshiue) [inline frontend/rust-lib/flowy-database2/src/services/calculations/controller.rs:245]: The calculation controller lives in the context of a view editor. Is there another reason why Self::view_id isn't used here?

File: frontend/rust-lib/flowy-database2/src/services/database_view/view_editor.rs [hunk 2]
```diff
@@ -1101,11 +1101,23 @@ impl DatabaseViewEditor {
...
+          .read()
+          .await
+          .did_create_row(row.clone().id)
+          .await;
+      }
+
+      if let Some(calculations_controller) = weak_calculations_controller.upgrade() {
+        calculations_controller
+          .did_receive_row_changed(row, Some(view_id))
+          .await;
```
Comment (richardshiue) [inline frontend/rust-lib/flowy-database2/src/services/database_view/view_editor.rs:1120]: You can pass the row as reference and only clone the row id in the sort controller

File: frontend/resources/translations/en.json [hunk 3]
```diff
@@ -739,8 +739,14 @@
...
-      "sum": "Sum"
-    }
+      "sum": "Sum",
+      "count": "Count",
+      "countEmpty": "Count empty",
+      "countEmptyShort": "Empty",
+      "countNonEmpty": "Count non empty",
+      "countNonEmptyShort": "Not empty"
+    },
+    "removeSorting": "Would you like to remove sorting?"
```
Comment (richardshiue) [inline frontend/resources/translations/en.json:749]: Oops

File: frontend/appflowy_flutter/integration_test/desktop/database/database_row_test.dart [hunk 4]
```diff
@@ -50,7 +49,6 @@ void main() {
 
       // The initial number of rows is 3
       await tester.assertNumberOfRowsInGridPage(2);
-      await tester.assertRowCountInGridPage(2);
       await tester.pumpAndSettle();
```
Comment (richardshiue) [inline frontend/appflowy_flutter/integration_test/desktop/database/database_row_test.dart:52]: Unrelated, but could the comment be fixed?

File: frontend/appflowy_flutter/integration_test/desktop/database/database_row_test.dart [hunk 5]
```diff
@@ -33,7 +33,6 @@ void main() {
 
       // The initial number of rows is 3
       await tester.assertNumberOfRowsInGridPage(4);
-      await tester.assertRowCountInGridPage(4);
       await tester.pumpAndSettle();
     });
```
Comment (richardshiue) [inline frontend/appflowy_flutter/integration_test/desktop/database/database_row_test.dart:38]: Unrelated, but could the comment be fixed?

File: frontend/appflowy_flutter/integration_test/desktop/database/database_row_test.dart (hunk 4 above)
Comment (Xazin) [reply frontend/appflowy_flutter/integration_test/desktop/database/database_row_test.dart:52]: What do you mean?

**Pull request merged**